### ISSUE TITLE
`geth export` tries to write on read-only datadir

### ISSUE THREAD

**Issue by alkorang** [ethereum/go-ethereum#29322]
#### System information

Geth version: `geth-1.13.14 version 1.13.14-stable-2bd6bd01`
CL client & version:`beacon-chain version Prysm/v5.0.1/a1a81d1720a0a3b850992d4825d0a023baa8e65a. Built at: 2024-03-08 20:31:40+00:00`
OS & Version: Linux

#### Expected behaviour

`geth export` exports blocks to a chain file without modifying the read-only mode datadir.

#### Actual behaviour

`geth export` terminates after trying to truncate extra state histories on the read-only mode datadir.

#### Steps to reproduce the behaviour

1. Sync the geth node with `--state.scheme path` option.
```
./geth --db.engine pebble --state.scheme path --datadir datadir.geth
```
2. Ctrl+C to terminate the geth node.
3. Run `geth export`
```
./geth export --db.engine pebble --state.scheme path --datadir datadir.geth 1-100000.chain 1 100000
```

#### Backtrace

````
INFO [03-24|10:33:55.878] Maximum peer count                       ETH=50 total=50
INFO [03-24|10:33:55.880] Using pebble as db engine
INFO [03-24|10:33:55.883] Set global gas cap                       cap=50,000,000
INFO [03-24|10:33:55.884] Initializing the KZG library             backend=gokzg
INFO [03-24|10:33:56.003] Using pebble as the backing database
INFO [03-24|10:33:56.003] Allocated cache and file handles         database=/home/alkorang/ethereum-geth-prysm/execution/datadir.geth/geth/chaindata cache=512.00MiB handles=524,288
INFO [03-24|10:33:57.582] Opened ancient database                  database=/home/alkorang/ethereum-geth-prysm/execution/datadir.geth/geth/chaindata/ancient/chain readonly=true
INFO [03-24|10:33:57.588] State scheme set by user                 scheme=path
INFO [03-24|10:33:57.589] Failed to load journal, discard it       err="unmatched journal want daeb87735849095c0e81fd38e9ddb5fa48f7b59874c02500d0022b5633d67a59 got e9db26f61c707a6d23b4af55e0d872cc9a58fa871458c5cc2d75badfb88edb13"
INFO [03-24|10:33:57.594] Opened ancient database                  database=/home/alkorang/ethereum-geth-prysm/execution/datadir.geth/geth/chaindata/ancient/state readonly=false
CRIT [03-24|10:33:57.594] Failed to truncate extra state histories err="pebble: read-only"
````

One workaround was running `geth import` with a small chain file (e.g. genesis block) before `geth export` to let the client fix state histories and compact DB offline.

**Comment by rjl493456442** [comment]
Are you sure you stopped the geth gracefully?

The log `INFO [03-24|10:33:57.589] Failed to load journal, discard it       err="unmatched journal want daeb87735849095c0e81fd38e9ddb5fa48f7b59874c02500d0022b5633d67a59 got e9db26f61c707a6d23b4af55e0d872cc9a58fa871458c5cc2d75badfb88edb13"
` indicates am unclean shutdown somehow. The extra state histories also relevant with the it.

**Comment by alkorang** [comment]
Thank you for the answer. I should send `SIGTERM` to gracefully stop my node like `killall geth` or `killall -TERM geth`.

I used to press Ctrl+C sending `SIGINT` to the node and no problem with the past versions. But it no more gracefully stops my node gracefully. When I googled, there were some recommendations to try `SIGINT`, `SIGHUP`, or `SIGABRT` signals, but all of them cannot gracefully stop my node.
